kroxylicious-filters/kroxylicious-oauthbearer-validation/src/main/java/io/kroxylicious/proxy/filter/oauthbearer/OauthBearerValidationFilter.java

```diff
@@ -159,6 +163,19 @@ public CompletionStage<ResponseFilterResult> onSaslAuthenticateResponse(short ap
                                                                             SaslAuthenticateResponseData response, FilterContext context) {
         if (response.errorCode() == NONE.code()) {
             this.validateAuthentication = false;
+            try {
+                SaslServer responseServer = Sasl.createSaslServer(OAUTHBEARER_MECHANISM, "kafka", null, null, this.oauthHandler);
+                try {
+                    responseServer.evaluateResponse(response.authBytes());
```

Comment:
Why are you feeding the response's auth bytes to the `SaslServer`? That doesn't make any sense.